drivers/volume/portworx/portworx.go

```diff
@@ -2736,6 +2768,40 @@ func (d *portworx) getPxctlStatus(n node.Node) (string, error) {
 	return api.Status_STATUS_NONE.String(), nil
 }
 
+func (d *portworx) upgradePortworxUsingOperator(stc *operatorcorev1.StorageCluster, url string, endpointVersion string) error {
+	imageTag := endpointVersion
+	if strings.Contains(url, "edge") {
+		url = fmt.Sprintf("%s/%s/version", url, endpointVersion)
+		stdout, err := exec.Command("curl", "-L", url).CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("error while getting px version url %s, error %v", url, err)
+		}
+		var data interface{}
+		err = yaml.Unmarshal(stdout, &data)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal px version. cause: %v", err)
+		}
+		versionMap := data.(map[string]interface{})
+		imageTag = versionMap["version"].(string)
+	}
+	stcImage := strings.Split(stc.Spec.Image, ":")[0]
+	stc.Spec.Image = fmt.Sprintf("%s:%s", stcImage, imageTag)
+
+	_, err := pxOperator.UpdateStorageCluster(stc)
+	if err != nil {
+		return fmt.Errorf("failed to update storagecluster. cause: %v", err)
+	}
+
+	logrus.Infof("Storage cluster updated to version: %s", endpointVersion)
+
+	for _, node := range node.GetStorageDriverNodes() {
```

Comment:
We should get rid of this and just call `ValidateStorageCluster` instead, because it also does this as well as many other checks and validations https://github.com/libopenstorage/operator/blob/3ddc52ecdb29baed3b5cc31a0375cea17ae66030/pkg/util/test/util.go#L362